CodeReviewStandards/rfc-008-review-checklist.md

```diff
@@ -0,0 +1,44 @@
+# Pull Request Standards
+
+## Summary
+
+A checklist designed to help facilitate code reviews by outlining a pre-review checklist to follow for the raiser of the pull request, and a reviewer checklist.
+
+## Problem
+
+Pull requests follow no set format and vary between projects, which in turn causes reviews to take longer and require more context switching. Code review and commentary often happens 'offline', which results in reduced context and fewer points of reference when reviewing changes later.
+
+## Proposal
+
+Before raising a pull request, the pull request SHOULD meet the following criteria:
+ - [ ] Pull request description outlines _what has changed_
+ - [ ] Pull request description outlines _why the change was made_
+ - [ ] If the project has a linter, it SHOULD not report any issues with the pull request
+ - [ ] Code follows language specific conventions
+ - [ ] Code follows project specific conventions
+ - [ ] Code has been refactored with quality and readability in mind
+ - [ ] Tests are obviously the product of practising TDD (where appropriate)
+ - [ ] Testing strategy meets all requirements
+ - [ ] There are high level feature/acceptance tests in place as appropriate
+ - [ ] Tests have been refactored for quality
+ - [ ] Tests all pass and the change satisfies the goal of _why the change was made_
+
+When reviewing a pull request, reviewer SHOULD consider at least the following questions:
+
+ - [ ] Does the code achieve the goal outlined in _why the change was made_?
+ - [ ] Could the change have been implemented using a different strategy that would be a credible alternative? If so, why would the alternative be better?
+ - [ ] Does the code follow all of the testing and standards requirements in the pre-review checklist?
+ - [ ] Is the code readable?
+ - [ ] Is the code maintainable?
+ - [ ] Are there any gaps or potential issues to be found in the testing strategy? Are the tests _sane_?
+
+Universal Pull Request Checklist - all pull requests MUST meet the following criteria
+
+- [ ] Before merging, summarise any offline discussion of review comments (why a change may or may not be implemented) to aid future reference
+- [ ] SHOULD avoid branching from branches, branch from origin/master where possible
+- [ ] SUGGEST commit history can be rebased into a single commit
```

Comment:
This is a little controversial I think. Don't think we have a common standard - maybe we should.

Personally, I like any commits which get merged into to be individually meaningful, but that doesn't mean only having one. Think there are situations where having a few commits is more meaningful than having a single feature level commit.

How about "SUGGEST commit history is rebased into meaningful commits"?